File: sportspace-frontend/src/models/entities/Facility.ts
import type { FacilitySchedule } from "./FacilitySchedule";
import type { Room } from "./Room";

export type Facility = {
  id: number;
  name: string;
  address: string;
  rooms: Room[];
  active: boolean;
  schedules: FacilitySchedule[];
};

File: sportspace/src/main/java/ee/margus/sportspace/entity/Facility.java
package ee.margus.sportspace.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Facility extends  BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Room> rooms;

    private Boolean active;

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FacilitySchedule> schedules;
}
